### PR TITLE
Skips signing when there is no upload URL

### PR DESCRIPTION
**Pull request by amznestebanpapp**
So we can run the scripts locally

### CODE REVIEW

**Review by AMZN-alexpete — COMMENTED** [review]
Will developers still be able to generate signed installers locally for testing without having to set CPACK_UPLOAD_URL?

**Comment by amznestebanpapp** [comment]
> Will developers still be able to generate signed installers locally for testing without having to set CPACK_UPLOAD_URL?

No, and we couldnt find a way with @amzn-changml to create those certificates locally. Once we figure out how to create those certificates properly, I can change this logic and use some other variable to disable/enable signing.
Using `CPACK_UPLOAD_URL` was the easy solution because locally that wont be indicated and likely if you are uploading you want it signed.

**Comment by amzn-changml** [comment]
> > Will developers still be able to generate signed installers locally for testing without having to set CPACK_UPLOAD_URL?
> 
> No, and we couldnt find a way with @amzn-changml to create those certificates locally. Once we figure out how to create those certificates properly, I can change this logic and use some other variable to disable/enable signing. Using `CPACK_UPLOAD_URL` was the easy solution because locally that wont be indicated and likely if you are uploading you want it signed.

Agreed with this. Specifically, people can create self-signed certificates and sign them, but will fail validation. If a developer chooses to ignore that, that's fine, but it breaks the try/catch logic in the signing workflow.

File: cmake/Platform/Windows/PackagingPreBuild.cmake [hunk 1]
```diff
@@ -6,21 +6,41 @@
...
 
-set(_signing_command
-    psexec.exe
-    -accepteula 
-    -nobanner 
-    -s
-    powershell.exe
+unset(_signing_command)
+find_program(_psiexec_path psexec.exe)
+if(_psiexec_path)
```
[the remark this answers is not in the document]
Comment (amzn-changml) [reply cmake/Platform/Windows/PackagingPreBuild.cmake:19]: To be clear, psexec is to allow breaking out of session 0 in a Jenkins run, but for running locally, it's not needed. This is specifically used to allow access to the signing certificate if private keys are stored in a HSM.

**Review by amzn-changml — APPROVED** [review]
Looks good. Future work might involve finding a way to pass validation with self-signed certificates, or create one at runtime.

**Review by AMZN-alexpete — APPROVED** [review]

File: cmake/Platform/Windows/PackagingPreBuild.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (AMZN-ScottR) [reply cmake/Platform/Windows/PackagingPreBuild.cmake:19]: Cool, thanks for the explanation

**Review by AMZN-ScottR — APPROVED** [review]

**Pull request merged**